src/Integration.Vsix/SonarLintTagger/AccumulatingIssueConsumer.cs

```diff
@@ -0,0 +1,117 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using SonarLint.VisualStudio.Core.Analysis;
+
+/* 
+ * Instancing: a new instance of this class should be created for each analysis request.
+ * 
+ * The class is initialized with the text snapshot representing the state of the text buffer
+ * at the point the analysis was triggered ("the analysis snaphost").
+ * 
+ * Each time IIssueConsumer.Accept is called, the new issues will be mapped back to the
+ * analysis snapshot and decorated with the additional data required for filtering and tagging.
+ * Then, all of the issues that have been received that habe been received so far will be
+ * passed to the OnIssuesChanged delegate.
+ * 
+ * However, it's possible that the text buffer could have been edited since the analysis
+ * was triggered. It is the responsibity of the callback to translate to supplied IssueMarkers
+ * to the current snapshot, if necessary.
+ * 
+ * 
+ * Mapping from reported issue line/char positions to the analysis snapshot
+ * ------------------------------------------------------------------------
+ * Currently, the JS/CFamily analyzers run against the file on disc, not the content of 
+ * the snapshot. We're making a couple of assumptions:
+ * 1) we're triggering analysis on save, so the state of the text buffer matches the file on disc, and
+ * 2) the file won't have had time to change again before the analyzer reads it so we expect always to
```

Comment:
"the file won't have had time to change again before the analyzer reads it" -- not necessarily true, but we would've cancelled the analysis and started a new one. 
Hm, maybe it's better to remove this line. And just say that we trigger on save, so the text buffer matches the file on disk at the time the analysis is triggered.